packages/jest-cli/src/reporters/JestReporterCoverage.js

```diff
@@ -0,0 +1,106 @@
+/**
+* Copyright (c) 2014-present, Facebook, Inc. All rights reserved.
+*
+* This source code is licensed under the BSD-style license found in the
+* LICENSE file in the root directory of this source tree. An additional grant
+* of patent rights can be found in the PATENTS file in the same directory.
+*
+* @flow
+*/
+'use strict';
+
+import type {Config} from 'types/Config';
+import type {AggregatedResult, TestResult} from 'types/TestResult';
+
+const JestReporterBase = require('./JestReporterBase');
+const chalk = require('chalk');
+const istanbul = require('istanbul');
+
+const FAIL_COLOR = chalk.bold.red;
+
+class JestReporterCoverage extends JestReporterBase {
+  _collector: istanbul.Collector;
+  _testCollectors: Object;
+
+  constructor() {
+    super();
+    this._collector = new istanbul.Collector();
+    this._testCollectors = Object.create(null);
+  }
+
+  onTestResult(
+    config: Config,
+    testResult: TestResult,
+    aggregatedResults: AggregatedResult,
+  ) {
+    if (config.collectCoverage && testResult.coverage) {
```

Comment:
we don't need the `config.collectCoverage` check any more, right? (We should still keep the other one, some results may not have coverage for whatever reason, and wouldn't wanna fail then).